dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -0,0 +1,204 @@
+package definitions
+
+import (
+	"errors"
+
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+)
+
+type RouteGroupValidator struct {
+	FilterRegistry filters.Registry
+}
+
+var defaultRouteGroupValidator = &RouteGroupValidator{}
+
+// validateRouteGroup validates a RouteGroupItem
+func ValidateRouteGroup(rg *RouteGroupItem) error {
+	return defaultRouteGroupValidator.Validate(rg)
+}
+
+func ValidateRouteGroupList(rgl RouteGroupList) error {
+	var err error
+	for _, rg := range rgl.Items {
+		err = errors.Join(err, defaultRouteGroupValidator.Validate(rg))
```

Comment:
We can use errors.Join only after 1.21 since it was introduced by 1.20
I've created https://github.com/zalando/skipper/issues/2488, see also https://github.com/zalando/skipper/issues/2433

Also existing code 
https://github.com/zalando/skipper/blob/78aabb52fe9fffbbf25dcbc1f35ae4fb43d1988a/dataclients/kubernetes/definitions/routegroups.go#L462-L470
seems to be wrong because [if err is nil errors.Wrap returns nil](https://github.com/pkg/errors/blob/5dd12d0cfe7f152f80558d591504ce685299311e/errors.go#L184-L187)

Reply:
should be resolved by https://github.com/zalando/skipper/pull/2492